sdp/processors/datasets/librispeech/create_initial_manifest.py

```diff
@@ -0,0 +1,138 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import fnmatch
+import glob
+import json
+import os
+import typing
+
+from sdp.processors.base_processor import BaseProcessor
+from sdp.utils.common import download_file, extract_archive
+
+
+def get_librispeech_url_list(names: list[str]) -> list[str]:
+    urls = [
+        "https://openslr.org/resources/12/dev-clean.tar.gz",
+        "https://openslr.org/resources/12/dev-other.tar.gz",
+        "https://openslr.org/resources/12/test-clean.tar.gz",
+        "https://openslr.org/resources/12/test-other.tar.gz",
+        "https://openslr.org/resources/12/train-clean-100.tar.gz",
+        "https://openslr.org/resources/12/train-clean-360.tar.gz",
+        "https://openslr.org/resources/12/train-other-500.tar.gz",
+    ]
+    if "all" not in names:
+        filtered_urls = [url for url in urls if url.split('/')[-1].split('.tar')[0] in names]
+    else:
+        filtered_urls = urls
+
+    if len(filtered_urls) == 0:
+        raise ValueError("No data found")
+
+    return filtered_urls
+
+
+class CreateInitialManifestLibrispeech(BaseProcessor):
+    """Processor to create initial manifest for the Librispeech dataset.
+
+    Dataset link: https://openslr.org/12
+
+    Will download all files, extract tars and create manifest file with the
+    'audio_filepath' and 'text' fields
```

Comment:
Small thing, please use double back-ticks instead of ', so these will look like code when viewed in the docs online

Reply:
done